components/num-util/src/fixed_decimal.rs

```diff
@@ -373,6 +377,135 @@ impl fmt::Display for FixedDecimal {
     }
 }
 
+impl FromStr for FixedDecimal {
+    type Err = Error;
+    fn from_str(input_str: &str) -> Result<Self, Self::Err> {
+        // input_str: the input string
+        // no_sign_str: the input string when the sign is removed from it
+        let no_sign_str: &str;
+        let mut is_negative = false;
+        if &input_str[0..1] == "-" {
+            is_negative = true;
+            no_sign_str = &input_str[1..];
+        } else {
+            no_sign_str = input_str;
+        }
+        // Compute length of each string once and store it, so if you use that multiple times,
+        //  you don't compute it multiple times
+        // has_dot: shows if your input has dot in it
+        // dot_index: gives the index of dot (after removing the sign)
+        let no_sign_str_len = no_sign_str.chars().count();
```

Comment:
`.chars().count()` is O(N).  You need it when the string might contain multi-byte characters, but since the string is expected to be ASCII, we can count bytes.  So, you can be more efficient with

```rust
let no_sign_str_len = no_sign_str.len();
```